components/skus/browser/rs/lib/src/sdk/credentials/fetch.rs

```diff
@@ -173,11 +306,83 @@ where
         let (resp, _) = request_with_retries.await?;
 
         let resp: Vec<ItemCredentialsResponse> = serde_json::from_slice(resp.body())?;
+        event!(Level::DEBUG, "done decoded body");
 
         let mut time_limited_creds: HashMap<String, Vec<TimeLimitedCredential>> = HashMap::new();
+        let mut time_limited_v2_creds: Vec::<String> = Vec::new();
 
-        for item_cred in resp {
+        'outer: for item_cred in resp {
             match item_cred {
+                ItemCredentialsResponse::TimeLimitedV2 {
+                    item_id,
+                    blinded_creds,
+                    signed_creds,
+                    batch_proof,
+                    public_key,
+                    issuer_id,
+                    valid_from,
+                    valid_to,
+                    ..
+                } => {
+                    if let Some(item_creds) =
+                        self.client.get_time_limited_v2_creds(&item_id).await?
+                    {
+                        // ignore
+                        let from = NaiveDateTime::parse_from_str(&valid_from, "%Y-%m-%dT%H:%M:%SZ")
+                                    .map_err(|_| {
+                                        InternalError::InvalidResponse(
+                                            "Could not parse valid from".to_string(),
+                                        )
+                                    })?;
+                        let to = NaiveDateTime::parse_from_str(&valid_to, "%Y-%m-%dT%H:%M:%SZ")
+                                    .map_err(|_| {
+                                        InternalError::InvalidResponse(
+                                            "Could not parse valid to".to_string(),
+                                        )
+                                    })?;
+                        for stored_cred in item_creds.creds.unwrap_or(Vec::<TimeLimitedV2Credential>::new()) {
```

Comment:
maybe something like:
```suggestion
                        if item_creds.creds.iter().find(|stored_cred| to == stored_cred.valid_to && from == stored_cred.valid_from).is_some() {
                            continue
                        }
```